subsys/bluetooth/shell/bt.c

```diff
@@ -1338,6 +1331,17 @@ static int cmd_advertise(const struct shell *sh, size_t argc, char *argv[])
 		}
 	}
 
+	atomic_clear(adv_opt);
```

Comment:
Why is `adv_opt` an atomic. And why is it a global? It does not seem to be used anywhere outside this function.

Reply:
The `adv_opt` is the shell adv options for the legacy advertiser (name can probably be improved). Since there can only be a single legacy advertiser, this is just a global value. For the extended advertising sets, we have an array called `adv_set_opt` that holds the options for each extended advertising set. 

The reason why it is global is probably just a leftover from an earlier version, and can seemingly be moved to a local variable. The same seems to go for the extended advertising sets. 

I also do not see any reason as to why it should be `atomic` as there should only ever be a single thread accessing it.

I suggest to fix these "issues" in a separate PR.